src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.CreateBookRequestDto;
+import com.example.onlinebookstore.mapper.BookMapper;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto bookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(bookRequestDto)));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto get(Long id) {
+        return bookMapper.toDto(bookRepository.get(id).orElseThrow(() ->
```

Comment:
This code is really hard to read. Also bad idea to throw RuntimeException
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Can't find book with id " + id));
        return bookMapper.toDto(book);
```